src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlDocletWriter.java

```diff
@@ -2151,4 +2213,152 @@ Script getMainBodyScript() {
     Content getVerticalSeparator() {
         return HtmlTree.SPAN(HtmlStyle.verticalSeparator, new FixedStringContent("|"));
     }
+
+    public void addPreviewSummary(Element forWhat, Content target) {
+        if (utils.isPreviewAPI(forWhat)) {
+            DocTree previewTree = utils.getPreviewTree(forWhat);
+            Content div = HtmlTree.DIV(HtmlStyle.block);
+            if (previewTree != null) {
+                div.add(HtmlTree.SPAN(HtmlStyle.previewLabel,
+                                      new RawHtml(utils.getPreviewTreeSummaryOrDetails(previewTree, true))));
+            } else {
+                div.add(HtmlTree.SPAN(HtmlStyle.previewLabel, contents.previewPhrase));
+            }
+            target.add(div);
+        }
+    }
+
+    public void addPreviewInfo(Element forWhat, Content target) {
+        if (utils.isPreviewAPI(forWhat)) {
+            //in Java platform:
+            HtmlTree previewDiv = HtmlTree.DIV(HtmlStyle.previewBlock);
+            DocTree previewTree = utils.getPreviewTree(forWhat);
+            if (previewTree != null) {
+                previewDiv.add(new HtmlTree(TagName.A).put(HtmlAttr.ID, "preview")
+                                                      .add(new RawHtml(utils.getPreviewTreeSummaryOrDetails(previewTree, false))));
```

Comment:
The `id` attribute needs to be unique within the page, so in addition to make the value not a valid java identifier (as @jonathan-gibbons pointed out in a comment elsewhere) we need to support multiple preview ids per page. One way to do this would be to add the element name to the id value, e.g. `preview-<element name>`.

Reply:
Of course the element name won't do for overloaded methods and constructors... `Links#getAnchor(ExecutableElement)` should be used for those.

Reply:
Uh, originally, there was only preview section per file, and I didn't fully realize the JDK's javadoc may have multiple such section. I'll work on this. Thanks for the pointer!